// angular-modal-gallery/dist/src/components/gallery.component.js
/*
 The MIT License (MIT)

 Copyright (c) 2017 Stefano Cappa (Ks89)

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Input, Output, EventEmitter, Component } from '@angular/core';
/**
 * Component with the gallery of thumbs.
 * In receives an array of Images and a boolean to show/hide
 * the gallery (feature used by imagePointer).
 * Also it emits click events as outputs.
 */
var GalleryComponent = (function () {
    function GalleryComponent() {
        this.show = new EventEmitter();
    }
    GalleryComponent.prototype.showModalGallery = function (index) {
        this.show.emit(index);
    };
    /**
     * Method to get `alt attribute`.
     * `alt` specifies an alternate text for an image, if the image cannot be displayed.
     * There is a similar version of this method into `modal-gallery.component.ts` that
     * receives an Image as input.
     * @param index Number that represents the image index.
     */
    GalleryComponent.prototype.getAltDescriptionByIndex = function (index) {
        if (!this.images) {
            return '';
        }
        if (!this.images[index] || !this.images[index].description) {
            return "Image " + index;
        }
        return this.images[index].description;
    };
    return GalleryComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Array)
], GalleryComponent.prototype, "images", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], GalleryComponent.prototype, "showGallery", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], GalleryComponent.prototype, "show", void 0);
GalleryComponent = __decorate([
    Component({
        selector: 'gallery',
        styleUrls: ['gallery.scss'],
        templateUrl: 'gallery.html'
    })
], GalleryComponent);
export { GalleryComponent };
