// angular-modal-gallery/aot/src/components/gallery.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from './gallery.scss.shim.ngstyle';
import * as i1 from '@angular/core';
import * as i2 from '@angular/common';
import * as i3 from '../../../src/components/gallery.component';
const styles_GalleryComponent:any[] = [i0.styles];
export const RenderType_GalleryComponent:i1.RendererType2 = i1.ɵcrt({encapsulation:0,
    styles:styles_GalleryComponent,data:{}});
function View_GalleryComponent_4(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),0,'img',[['class',
      'ng-thumb']],[[8,'src',4],[8,'alt',0]],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.showModalGallery((<any>(<any>_v.parent).parent).context.index)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any)))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = i1.ɵinlineInterpolate(1,'',(<any>(<any>_v.parent).parent).context.$implicit.thumb,
        '');
    const currVal_1:any = i1.ɵinlineInterpolate(1,'',_co.getAltDescriptionByIndex((<any>(<any>_v.parent).parent).context.index),
        '');
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
function View_GalleryComponent_5(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),0,'img',[['class',
      'ng-thumb']],[[8,'src',4],[8,'alt',0]],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.showModalGallery((<any>(<any>_v.parent).parent).context.index)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any)))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = i1.ɵinlineInterpolate(1,'',(<any>(<any>_v.parent).parent).context.$implicit.img,
        '');
    const currVal_1:any = i1.ɵinlineInterpolate(1,'',_co.getAltDescriptionByIndex((<any>(<any>_v.parent).parent).context.index),
        '');
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
function View_GalleryComponent_3(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),7,(null as any),
      (null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i1.ɵted((null as any),['\n      '])),(_l()(),i1.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_GalleryComponent_4)),i1.ɵdid(16384,(null as any),
          0,i2.NgIf,[i1.ViewContainerRef,i1.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i1.ɵted((null as any),['\n      '])),(_l()(),i1.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_GalleryComponent_5)),i1.ɵdid(16384,(null as any),
          0,i2.NgIf,[i1.ViewContainerRef,i1.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i1.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    const currVal_0:any = (<any>_v.parent).context.$implicit.thumb;
    _ck(_v,3,0,currVal_0);
    const currVal_1:boolean = !(<any>_v.parent).context.$implicit.thumb;
    _ck(_v,6,0,currVal_1);
  },(null as any));
}
function View_GalleryComponent_2(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),4,'div',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i1.ɵted((null as any),['\n    '])),(_l()(),i1.ɵand(16777216,(null as any),(null as any),
      1,(null as any),View_GalleryComponent_3)),i1.ɵdid(16384,(null as any),0,i2.NgIf,
      [i1.ViewContainerRef,i1.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
      i1.ɵted((null as any),['\n  ']))],(_ck,_v) => {
    const currVal_0:any = (_v.context.$implicit && _v.context.$implicit.img);
    _ck(_v,3,0,currVal_0);
  },(null as any));
}
function View_GalleryComponent_1(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),4,'div',[['class',
      'ng-gallery']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i1.ɵted((null as any),['\n  '])),(_l()(),i1.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_GalleryComponent_2)),i1.ɵdid(802816,(null as any),
          0,i2.NgForOf,[i1.ViewContainerRef,i1.TemplateRef,i1.IterableDiffers],{ngForOf:[0,
              'ngForOf']},(null as any)),(_l()(),i1.ɵted((null as any),['\n']))],(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.images;
    _ck(_v,3,0,currVal_0);
  },(null as any));
}
export function View_GalleryComponent_0(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_GalleryComponent_1)),i1.ɵdid(16384,(null as any),0,i2.NgIf,[i1.ViewContainerRef,
      i1.TemplateRef],{ngIf:[0,'ngIf']},(null as any))],(_ck,_v) => {
    var _co:i3.GalleryComponent = _v.component;
    const currVal_0:any = _co.showGallery;
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
export function View_GalleryComponent_Host_0(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),1,'gallery',([] as any[]),
      (null as any),(null as any),(null as any),View_GalleryComponent_0,RenderType_GalleryComponent)),
      i1.ɵdid(49152,(null as any),0,i3.GalleryComponent,([] as any[]),(null as any),
          (null as any))],(null as any),(null as any));
}
export const GalleryComponentNgFactory:i1.ComponentFactory<i3.GalleryComponent> = i1.ɵccf('gallery',
    i3.GalleryComponent,View_GalleryComponent_Host_0,{images:'images',showGallery:'showGallery'},
    {show:'show'},([] as any[]));
