// angular-modal-gallery/aot/src/components/modal-gallery.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from './modal-gallery.scss.shim.ngstyle';
import * as i1 from '@angular/core';
import * as i2 from '../../../src/directives/click-outside.directive';
import * as i3 from '@angular/common';
import * as i4 from './upper-buttons.component.ngfactory';
import * as i5 from '../../../src/components/upper-buttons.component';
import * as i6 from '../../../src/components/modal-gallery.component';
import * as i7 from './gallery.component.ngfactory';
import * as i8 from '../../../src/components/gallery.component';
import * as i9 from '../../../src/services/keyboard.service';
const styles_AngularModalGalleryComponent:any[] = [i0.styles];
export const RenderType_AngularModalGalleryComponent:i1.RendererType2 = i1.ɵcrt({encapsulation:0,
    styles:styles_AngularModalGalleryComponent,data:{}});
function View_AngularModalGalleryComponent_2(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),3,'div',[['class',
      'uil-ring-css']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i1.ɵted((null as any),['\n        '])),(_l()(),i1.ɵeld(0,(null as any),
          (null as any),0,'div',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i1.ɵted((null as any),['\n      ']))],
      (null as any),(null as any));
}
function View_AngularModalGalleryComponent_3(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),2,'a',[['class',
      'nav-left']],[[8,'hidden',0]],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.prevImage()) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i1.ɵeld(0,(null as any),(null as any),0,
      'i',[['class','fa fa-angle-left']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i1.ɵted((null as any),['\n      ']))],
      (null as any),(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.isFirstImage;
        _ck(_v,0,0,currVal_0);
      });
}
function View_AngularModalGalleryComponent_4(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),0,'img',[['class',
      'effect']],[[8,'src',4],[8,'alt',0]],[[(null as any),'click'],[(null as any),
      'swipeleft'],[(null as any),'swiperight']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.nextImage(_co.clickAction)) !== false);
      ad = (pd_0 && ad);
    }
    if (('swipeleft' === en)) {
      const pd_1:any = ((<any>_co.swipe(_co.currentImageIndex,$event.type)) !== false);
      ad = (pd_1 && ad);
    }
    if (('swiperight' === en)) {
      const pd_2:any = ((<any>_co.swipe(_co.currentImageIndex,$event.type)) !== false);
      ad = (pd_2 && ad);
    }
    return ad;
  },(null as any),(null as any)))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = i1.ɵinlineInterpolate(1,'',_co.currentImage.img,'');
    const currVal_1:any = i1.ɵinlineInterpolate(1,'',_co.getAltDescriptionByImage(_co.currentImage),
        '');
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
function View_AngularModalGalleryComponent_5(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),2,'a',[['class',
      'nav-right']],[[8,'hidden',0]],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.nextImage()) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i1.ɵeld(0,(null as any),(null as any),0,
      'i',[['class','fa fa-angle-right']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i1.ɵted((null as any),['\n      ']))],
      (null as any),(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.isLastImage;
        _ck(_v,0,0,currVal_0);
      });
}
function View_AngularModalGalleryComponent_1(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),23,'div',[['class',
      'ng-overlay']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i1.ɵted((null as any),['\n\n  '])),(_l()(),i1.ɵeld(0,(null as any),(null as any),
          20,'div',[['class','ng-gallery-content'],['click-outside',''],['id','ng-gallery-content']],
          (null as any),[[(null as any),'clickOutside'],['document','click']],(_v,
              en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('document:click' === en)) {
              const pd_0:any = ((<any>i1.ɵnov(_v,3).onClick($event.target)) !== false);
              ad = (pd_0 && ad);
            }
            if (('clickOutside' === en)) {
              const pd_1:any = ((<any>_co.onClickOutside($event)) !== false);
              ad = (pd_1 && ad);
            }
            return ad;
          },(null as any),(null as any))),i1.ɵdid(16384,(null as any),0,i2.ClickOutsideDirective,
          ([] as any[]),{clickOutsideEnable:[0,'clickOutsideEnable']},{clickOutside:'clickOutside'}),
      (_l()(),i1.ɵted((null as any),['\n\n      '])),(_l()(),i1.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_AngularModalGalleryComponent_2)),i1.ɵdid(16384,
          (null as any),0,i3.NgIf,[i1.ViewContainerRef,i1.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i1.ɵted((null as any),['\n\n      '])),(_l()(),i1.ɵeld(0,
          (null as any),(null as any),1,'upperButtons',([] as any[]),(null as any),
          [[(null as any),'close'],[(null as any),'download']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('close' === en)) {
              const pd_0:any = ((<any>_co.closeGallery()) !== false);
              ad = (pd_0 && ad);
            }
            if (('download' === en)) {
              const pd_1:any = ((<any>_co.downloadImage()) !== false);
              ad = (pd_1 && ad);
            }
            return ad;
          },i4.View_UpperButtonsComponent_0,i4.RenderType_UpperButtonsComponent)),
      i1.ɵdid(49152,(null as any),0,i5.UpperButtonsComponent,([] as any[]),{image:[0,
          'image'],configButtons:[1,'configButtons']},{close:'close',download:'download'}),
      (_l()(),i1.ɵted((null as any),['\n\n      '])),(_l()(),i1.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_AngularModalGalleryComponent_3)),i1.ɵdid(16384,
          (null as any),0,i3.NgIf,[i1.ViewContainerRef,i1.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i1.ɵted((null as any),['\n      '])),(_l()(),i1.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_AngularModalGalleryComponent_4)),
      i1.ɵdid(16384,(null as any),0,i3.NgIf,[i1.ViewContainerRef,i1.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i1.ɵted((null as any),['\n      '])),(_l()(),
          i1.ɵand(16777216,(null as any),(null as any),1,(null as any),View_AngularModalGalleryComponent_5)),
      i1.ɵdid(16384,(null as any),0,i3.NgIf,[i1.ViewContainerRef,i1.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i1.ɵted((null as any),['\n      '])),(_l()(),
          i1.ɵeld(0,(null as any),(null as any),1,'span',[['class','info-text']],(null as any),
              (null as any),(null as any),(null as any),(null as any))),(_l()(),i1.ɵted((null as any),
          ['',''])),(_l()(),i1.ɵted((null as any),['\n  '])),(_l()(),i1.ɵted((null as any),
          ['\n']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.enableCloseOutside;
    _ck(_v,3,0,currVal_0);
    const currVal_1:any = _co.loading;
    _ck(_v,6,0,currVal_1);
    const currVal_2:any = _co.currentImage;
    const currVal_3:any = _co.configButtons;
    _ck(_v,9,0,currVal_2,currVal_3);
    const currVal_4:any = (((_co.images == null)? (null as any): _co.images.length) > 1);
    _ck(_v,12,0,currVal_4);
    const currVal_5:boolean = !_co.loading;
    _ck(_v,15,0,currVal_5);
    const currVal_6:any = (((_co.images == null)? (null as any): _co.images.length) > 1);
    _ck(_v,18,0,currVal_6);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_7:any = _co.getDescriptionToDisplay();
    _ck(_v,21,0,currVal_7);
  });
}
export function View_AngularModalGalleryComponent_0(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),1,'gallery',([] as any[]),
      (null as any),[[(null as any),'show']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i6.AngularModalGalleryComponent = _v.component;
        if (('show' === en)) {
          const pd_0:any = ((<any>_co.onShowModalGallery($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },i7.View_GalleryComponent_0,i7.RenderType_GalleryComponent)),i1.ɵdid(49152,
      (null as any),0,i8.GalleryComponent,([] as any[]),{images:[0,'images'],showGallery:[1,
          'showGallery']},{show:'show'}),(_l()(),i1.ɵted((null as any),['\n\n'])),
      (_l()(),i1.ɵand(16777216,(null as any),(null as any),1,(null as any),View_AngularModalGalleryComponent_1)),
      i1.ɵdid(16384,(null as any),0,i3.NgIf,[i1.ViewContainerRef,i1.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any))],(_ck,_v) => {
    var _co:i6.AngularModalGalleryComponent = _v.component;
    const currVal_0:any = _co.images;
    const currVal_1:any = _co.showGallery;
    _ck(_v,1,0,currVal_0,currVal_1);
    const currVal_2:any = _co.opened;
    _ck(_v,4,0,currVal_2);
  },(null as any));
}
export function View_AngularModalGalleryComponent_Host_0(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),1,'modal-gallery',
      ([] as any[]),(null as any),[['window','keydown']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('window:keydown' === en)) {
          const pd_0:any = ((<any>i1.ɵnov(_v,1).onKeyDown($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_AngularModalGalleryComponent_0,RenderType_AngularModalGalleryComponent)),
      i1.ɵdid(770048,(null as any),0,i6.AngularModalGalleryComponent,[i9.KeyboardService],
          (null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const AngularModalGalleryComponentNgFactory:i1.ComponentFactory<i6.AngularModalGalleryComponent> = i1.ɵccf('modal-gallery',
    i6.AngularModalGalleryComponent,View_AngularModalGalleryComponent_Host_0,{modalImages:'modalImages',
        imagePointer:'imagePointer',downloadable:'downloadable',description:'description',
        buttonsConfig:'buttonsConfig',keyboardConfig:'keyboardConfig',enableCloseOutside:'enableCloseOutside',
        slideConfig:'slideConfig',showDownloadButton:'showDownloadButton',showExtUrlButton:'showExtUrlButton'},
    {close:'close',show:'show',firstImage:'firstImage',lastImage:'lastImage',hasData:'hasData'},
    ([] as any[]));
