// ng2-image-gallery/dist/src/ng2-image-gallery.component.js
import { Component, Input, EventEmitter, Output, Inject, forwardRef } from '@angular/core';
import { DragulaService } from 'ng2-dragula';
var Ng2ImageGalleryComponent = (function () {
    function Ng2ImageGalleryComponent(dragulaService) {
        var _this = this;
        this.dragulaService = dragulaService;
        this.images = [];
        this.draggable = false;
        this.asImage = "image";
        this.asThumbnail = "thumbnail";
        this.asText = "text";
        this.actionText = "";
        this.actionEmitter = new EventEmitter();
        this.dropEmitter = new EventEmitter();
        this.loading = false;
        this.curImageIndex = 0;
        this.curThumbnailIndex = 0;
        this.isLightboxOpen = false;
        dragulaService.drop.subscribe(function (value) {
            _this.onDrop(value.slice(1));
        });
    }
    Ng2ImageGalleryComponent.prototype.ngOnChanges = function (changes) {
        if (changes.images) {
            this.images = changes.images.currentValue;
        }
    };
    Ng2ImageGalleryComponent.prototype.openLightboxGallery = function (index) {
        var _this = this;
        this.isLightboxOpen = true;
        setTimeout(function () {
            _this.curImageIndex = index - 1;
            _this.nextImage();
            _this.loading = false;
        }, 0);
    };
    Ng2ImageGalleryComponent.prototype.closeLightboxGallery = function () {
        this.isLightboxOpen = false;
    };
    Ng2ImageGalleryComponent.prototype.nextImage = function () {
        if (this.curImageIndex !== this.images.length - 1) {
            this.loading = true;
            this.curImageIndex++;
            this.curThumbnailIndex = this.curImageIndex;
        }
    };
    Ng2ImageGalleryComponent.prototype.previousImage = function () {
        this.loading = true;
        if (this.curImageIndex !== 0) {
            this.curImageIndex--;
        }
        this.curThumbnailIndex = this.curImageIndex;
    };
    Ng2ImageGalleryComponent.prototype.toImage = function (index) {
        this.loading = true;
        this.curImageIndex = index;
        this.curThumbnailIndex = this.curImageIndex;
    };
    Ng2ImageGalleryComponent.prototype.thumbnailsNext = function () {
        if (this.curThumbnailIndex + 5 < this.images.length) {
            this.curThumbnailIndex += 5;
        }
        else {
            this.curThumbnailIndex = this.images.length - 1;
        }
        this.curThumbnailIndex += 5;
    };
    Ng2ImageGalleryComponent.prototype.thumbnailsPrevious = function () {
        if (this.curThumbnailIndex - 5 > 0) {
            this.curThumbnailIndex -= 5;
        }
        else {
            this.curThumbnailIndex = 0;
        }
    };
    Ng2ImageGalleryComponent.prototype.onLoad = function () {
        this.loading = false;
    };
    Ng2ImageGalleryComponent.prototype.onAction = function (image) {
        this.actionEmitter.emit({
            image: image
        });
    };
    Ng2ImageGalleryComponent.prototype.onDrop = function (args) {
        var e = args[0], el = args[1];
        this.dropEmitter.emit({
            images: this.images
        });
    };
    return Ng2ImageGalleryComponent;
}());
export { Ng2ImageGalleryComponent };
Ng2ImageGalleryComponent.decorators = [
    { type: Component, args: [{
                selector: 'ng2-image-gallery',
                template: "\n    <div class=\"ng2-image-gallery-thumbnails\" *ngIf=\"!draggable\">\n        <div *ngFor=\"let image of images; let index=index\" class=\"ng2-image-gallery-thumbnail-container\">\n            <img [src]=\"image[asThumbnail]\" (click)=\"openLightboxGallery(index)\" class=\"ng2-image-gallery-thumbnail\">\n            <button class=\"btn btn-primary ng2-image-gallery-action\" *ngIf=\"actionText!==''\" (click)=\"onAction(image)\" [innerHTML]=\"actionText\"></button>\n        </div>\n    </div>\n    <div class=\"ng2-image-gallery-thumbnails\" [dragula]=\"'ng2-image-gallery-bag'\" [dragulaModel]='images' *ngIf=\"draggable\">\n        <div *ngFor=\"let image of images; let index=index\" class=\"ng2-image-gallery-thumbnail-container\">\n            <img [src]=\"image[asThumbnail]\" (click)=\"openLightboxGallery(index)\" class=\"ng2-image-gallery-thumbnail\">\n            <button class=\"btn btn-primary ng2-image-gallery-action\" *ngIf=\"actionText!==''\" (click)=\"onAction(image)\" [innerHTML]=\"actionText\"></button>\n        </div>\n    </div>\n    <to-body-displacer>\n        <div class=\"ng2-image-gallery-overlay\" [hidden]=\"!isLightboxOpen\">\n            <div class=\"ng2-image-gallery-lightbox\">\n                <a class=\"ng2-igl-close\" (click)=\"closeLightboxGallery()\"><i class=\"fa fa-close\"></i></a>\n                <a class=\"ng2-igl-nav-left\" *ngIf=\"images.length >1 && curImageIndex!==0\" (click)=\"previousImage()\"><i class=\"fa fa-angle-left\"></i></a>\n                <img [hidden]=\"loading\" *ngIf=\"images.length>0\" [src]=\"images[curImageIndex][asImage]\" (click)=\"nextImage()\" class=\"ng2-igl-img\" (load)=\"onLoad()\" />\n                <span [hidden]=\"!loading\" *ngIf=\"images.length>0\" class=\"ng2-igl-loading\"><i class=\"fa fa-spinner fa-spin fa-3x\" aria-hidden=\"true\"></i></span>\n                <a class=\"ng2-igl-nav-right\" *ngIf=\"images.length >1 && curImageIndex!==images.length-1\" (click)=\"nextImage()\"><i class=\"fa fa-angle-right\"></i></a>\n                <span class=\"ng2-igl-text\" *ngIf=\"images.length>0\" [innerHTML]=\"images[curImageIndex][asText]\"></span>\n                <span class=\"ng2-igl-count\">{{curImageIndex+1}}/{{images.length}}</span>\n                <div class=\"ng2-igl-thumbnails\">\n                    <a class=\"ng2-igl-nav-left thumbnails\" [ngClass]=\"{'non-visible': curThumbnailIndex < 4 }\" (click)=\"thumbnailsPrevious()\"><i class=\"fa fa-angle-left\"></i></a>\n                    <ng-template ngFor let-image [ngForOf]=\"images\" let-index=\"index\">\n                        <img [src]=\"image[asThumbnail]\" (click)=\"toImage(index)\" \n                            *ngIf=\"(curThumbnailIndex>images.length-4 && index > images.length-8)||(curThumbnailIndex<4 && index < 7)||(index >= curThumbnailIndex -3) && (index <= curThumbnailIndex+3)\"\n                            [ngClass]=\"{'active':curImageIndex===index}\"\n                            class=\"ng2-igl-thumbnail rounded-circle\">\n                    </ng-template>\n                    <a class=\"ng2-igl-nav-right thumbnails\" [ngClass]=\"{'non-visible': curThumbnailIndex > images.length-5 }\" (click)=\"thumbnailsNext()\"><i class=\"fa fa-angle-right\"></i></a>\n                </div>\n            </div>\n        </div>\n    </to-body-displacer>"
            },] },
];
/** @nocollapse */
Ng2ImageGalleryComponent.ctorParameters = function () { return [
    { type: DragulaService, decorators: [{ type: Inject, args: [forwardRef(function () { return DragulaService; }),] },] },
]; };
Ng2ImageGalleryComponent.propDecorators = {
    'images': [{ type: Input },],
    'draggable': [{ type: Input, args: ['draggable',] },],
    'asImage': [{ type: Input, args: ['asImage',] },],
    'asThumbnail': [{ type: Input, args: ['asThumbnail',] },],
    'asText': [{ type: Input, args: ['asText',] },],
    'actionText': [{ type: Input, args: ['actionText',] },],
    'actionEmitter': [{ type: Output, args: ['onAction',] },],
    'dropEmitter': [{ type: Output, args: ['onDrop',] },],
};
