// ng-scrollreveal/ngs-reveal.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { WindowService, NgsRevealService, NgsRevealConfig } from './services/index';
import { NgsRevealDirective, NgsRevealSetDirective } from './directives/index';
export { WindowService, NgsRevealService, NgsRevealConfig } from './services/index';
export { NgsRevealDirective, NgsRevealSetDirective } from './directives/index';
/**
 * Main module of the library
 */
export var NgsRevealModule = (function () {
    function NgsRevealModule() {
    }
    NgsRevealModule.forRoot = function () {
        return {
            ngModule: NgsRevealModule,
            providers: [WindowService, NgsRevealService, NgsRevealConfig]
        };
    };
    NgsRevealModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    exports: [NgsRevealDirective, NgsRevealSetDirective],
                    declarations: [NgsRevealDirective, NgsRevealSetDirective]
                },] },
    ];
    /** @nocollapse */
    NgsRevealModule.ctorParameters = [];
    return NgsRevealModule;
}());
