// angular-modal-gallery/dist/src/modal-gallery.module.js
/*
 The MIT License (MIT)

 Copyright (c) 2017 Stefano Cappa (Ks89)

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 */
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { NgModule, InjectionToken } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AngularModalGalleryComponent } from './components/modal-gallery.component';
import { DIRECTIVES } from './directives/directives';
import { UpperButtonsComponent } from './components/upper-buttons.component';
import { KeyboardService, } from './services/keyboard.service';
import { GalleryComponent } from './components/gallery.component';
export var KEYBOARD_CONFIGURATION = new InjectionToken('KEYBOARD_CONFIGURATION');
/**
 * Module with `forRoot` method to import it in the root module of your application.
 */
var ModalGalleryModule = ModalGalleryModule_1 = (function () {
    function ModalGalleryModule() {
    }
    ModalGalleryModule.forRoot = function (config) {
        return {
            ngModule: ModalGalleryModule_1,
            providers: [
                {
                    provide: KEYBOARD_CONFIGURATION,
                    useValue: config ? config : {}
                },
                {
                    provide: KeyboardService,
                    useFactory: setupRouter,
                    deps: [KEYBOARD_CONFIGURATION]
                }
            ]
        };
    };
    return ModalGalleryModule;
}());
ModalGalleryModule = ModalGalleryModule_1 = __decorate([
    NgModule({
        imports: [CommonModule],
        declarations: [AngularModalGalleryComponent, UpperButtonsComponent, GalleryComponent, DIRECTIVES],
        exports: [AngularModalGalleryComponent]
    })
], ModalGalleryModule);
export { ModalGalleryModule };
export function setupRouter(injector) {
    return new KeyboardService(injector);
}
var ModalGalleryModule_1;
